frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -138,25 +139,33 @@ export const startPipeline: KebabAction = (
   },
 });
 
-export const rerunPipeline: KebabAction = (
-  kind: K8sKind,
-  pipeline: Pipeline,
-  latestRun: PipelineRun,
-  onSubmit?: (pipelineRun: PipelineRun) => void,
-) => {
+export const rerunPipeline: KebabAction = (kind: K8sKind, pipelineRun: PipelineRun) => {
   // The returned function will be called using the 'kind' and 'obj' in Kebab Actions
   return {
     label: 'Start Last Run',
     callback: () => {
-      k8sCreate(PipelineRunModel, getPipelineRunData(pipeline, latestRun))
-        .then(onSubmit)
+      k8sCreate(
+        PipelineRunModel,
+        getPipelineRunData(
+          {
+            apiVersion: `${PipelineModel.apiGroup}/${PipelineModel.apiVersion}`,
+            kind: 'Pipeline',
+            metadata: {
+              name: pipelineRun.spec.pipelineRef.name,
+              namespace: pipelineRun.metadata.namespace,
+            },
+          },
```

Comment:
Everything we need is already in the Pipeline Run. Could you please refactor / update / create-a-new this function so that all you need to do is pass in pipeline run itself instead of fabricating a fake Pipeline just to later extract the data?

Reply:
Yes. Updated existing function.